src/View/SSViewer_DataPresenter.php

```diff
@@ -4,14 +4,15 @@
 
 use InvalidArgumentException;
 use SilverStripe\Core\ClassInfo;
+use SilverStripe\Dev\Deprecation;
 use SilverStripe\ORM\FieldType\DBField;
 
 /**
  * This extends SSViewer_Scope to mix in data on top of what the item provides. This can be "global"
  * data that is scope-independant (like BaseURL), or type-specific data that is layered on top cross-cut like
  * (like $FirstLast etc).
  *
- * It's separate from SSViewer_Scope to keep that fairly complex code as clean as possible.
+ * @deprecated 5.4.0 Will be merged into SilverStripe\View\SSViewer_Scope
  */
 class SSViewer_DataPresenter extends SSViewer_Scope
```

Comment:
I know this was intentionally kept separate, but whatever the case was when that decision was made it was causing me headaches trying to sort out what the difference is between `SSViewer_Scope` and this class. Especially since they have a lot of similar logic in them - the `__call()` class is a great example where the code flow is _not_ obvious with this class split out like this.

I've opted to merge the two classes together, and simplify where I can. There are or will be additional cards to simplify it further in the future, too.